tests/specs/nodeRewardsProgram.spec.js

```diff
@@ -0,0 +1,373 @@
+            mockFetch([{
+                recipientAddress: 'NCHESTYVD2P6P646AMY7WSNG73PCPZDUQNSD6JAK',
+                transactionHeight: 1,
+                transactionHash: 'bba40188dc1f042da959309639a884d8f6a87086cda10300d2a7c3a0e0891aua',
+                host: 'http://108.61.168.86:7890'
+            }]);
+            
+            const publicKey = '00a30788dc1f042da959309639a884d8f6a87086cda10300d2a7c3a0e0891a4d';
+            const expectedEnrollmentStatus = true;
+
+            // Act:
+            const enrollmentStatus = await NodeRewardsProgram.checkEnrollmentStatus(publicKey);
+
+            // Assert:
+            expect(enrollmentStatus).toBe(expectedEnrollmentStatus);
+            done();
+        });
+
+        it('should return false when there is successfull enrollment for given public key and recipient address is not a valid enrollment address', async (done) => {
+            // Arrange:
+            spyOn(NodeRewardsProgram, 'checkEnrollmentAddress').and.returnValue(Promise.resolve(false));
+            mockFetch([{
+                recipientAddress: 'NCHESTYVD2P6P646AMY7WSNG73PCPZDUQNSD6JAK',
+                transactionHeight: 1,
+                transactionHash: 'bba40188dc1f042da959309639a884d8f6a87086cda10300d2a7c3a0e0891aua',
+                host: 'http://108.61.168.86:7890'
+            }]);
+            const publicKey = '00a30788dc1f042da959309639a884d8f6a87086cda10300d2a7c3a0e0891a4d';
+            const expectedEnrollmentStatus = false;
+
+            // Act:
+            const enrollmentStatus = await NodeRewardsProgram.checkEnrollmentStatus(publicKey);
+
+            // Assert:
+            expect(enrollmentStatus).toBe(expectedEnrollmentStatus);
+            done();
+        });
+
+        it('should return false when there is no successfull enrollment for given public key', async (done) => {
+            // Arrange:
+            mockFetch([]);
+            const publicKey = '00a30788dc1f042da959309639a884d8f6a87086cda10300d2a7c3a0e0891a4d';
+            const expectedEnrollmentStatus = false;
+
+            // Act:
+            const enrollmentStatus = await NodeRewardsProgram.checkEnrollmentStatus(publicKey);
+
+            // Assert:
+            expect(enrollmentStatus).toBe(expectedEnrollmentStatus);
+            done();
+        });
+
+        it('should throw an error when request failed', async (done) => {
+            // Arrange:
+            mockFetch(null, false);
+            const publicKey = '00a30788dc1f042da959309639a884d8f6a87086cda10300d2a7c3a0e0891a4d';
+            const expectedError = Error('failed_to_get_success_enrollments');
+
+            // Act:
+            try {
+                await NodeRewardsProgram.checkEnrollmentStatus(publicKey);
+            }
+            catch(e) {
+                // Assert:
+                expect(e.message).toBe(expectedError.message);
+                done();
+            }
+        });
+    });
```

Comment:
you could also write helper for error cases since pattern is same across all

Reply:
Added helper function.